### PR TITLE
fix snapshot generation hanging on windows

### PR DESCRIPTION
**Pull request by jakemac53**
Fixes https://github.com/dart-lang/build/issues/2998

Logs at `fine` level everything on stdout, and `warning` everything on stderr. Adds an info level message if anything is visible on stdout telling you to run in `--verbose` mode.

This fixes the issue by actually draining the stdout stream. I am unsure why this is only presenting itself as an issue on windows, but seems like a good change anyways.

### CODE REVIEW

File: build_runner/lib/src/build_script_generate/bootstrap.dart [hunk 1]
```diff
@@ -156,27 +156,40 @@ Future<int> _createSnapshotIfNeeded(Logger logger) async {
...
+            .transform(LineSplitter())
+            .listen((l) {
+          hadStdOut = true;
+          logger.fine('stdout: $l');
+        }).asFuture(),
+      ]);
     });
+    if (hadStdOut) {
+      logger.info('There was output on stdout while compiling the build script '
+          'snapshot, run with `--verbose` to see the it\n.');
```
Comment (natebosch) [inline build_runner/lib/src/build_script_generate/bootstrap.dart:186]: Will they need to do a clean first or something? If it was successful but with output I don't think the next run with rebuild the snapshot will it?

Comment (jakemac53) [reply build_runner/lib/src/build_script_generate/bootstrap.dart:186]: Hmmm right, they would have to clean. Should we suggest a `clean` do you think? Just not emit a message?

Comment (natebosch) [reply build_runner/lib/src/build_script_generate/bootstrap.dart:186]: I guess maybe suggest a `clean` for now, and file an issue to add a `--force-snapshot` flag.

Comment (jakemac53) [reply build_runner/lib/src/build_script_generate/bootstrap.dart]: I don't think we actually want to add a `--force-snapshot` flag - just a normal clean in this scenario should be fine IMO.

File: build_runner/lib/src/build_script_generate/bootstrap.dart [hunk 2]
```diff
@@ -156,27 +156,41 @@ Future<int> _createSnapshotIfNeeded(Logger logger) async {
...
+            .transform(LineSplitter())
+            .listen((l) {
+          hadStdOut = true;
+          logger.fine('stdout: $l');
+        }).asFuture(),
+      ]);
     });
+    if (hadStdOut) {
+      logger.info('There was output on stdout while compiling the build script '
+          'snapshot, run with `--verbose` to see the it (you will need to run '
```
Comment (natebosch) [inline build_runner/lib/src/build_script_generate/bootstrap.dart:186]:
```suggestion
          'snapshot, run with `--verbose` to see it (you will need to run '
```

**Review by natebosch — APPROVED** [review]

**Comment by jakemac53** [comment]
The test failure here I am consistently seeing on master now as well. There is something broken with `dev`, but I am going to go ahead and merge this and try to address that separately.

**Pull request merged**